### PR TITLE
:memo: Add contributing guide

### PR DESCRIPTION
**Pull request by nefrob**
# Make contributing easier

### Changes
- Adds contributing guide
- Makes required `tox` version consistent across files
- Adds command to lint files to `tox.ini`
- Adds command to regenerate dependency files 

### Action items
- [ ] add re-compile dependencies command
- [ ] clean up where lint and compile dependencies live

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+4. Lint your code via pre-commit:
+
+    ```shell
+    tox -e lint
+    ```
+
+5. Re-compile dependencies:
+
+    ```shell
+    tox TODO
```
Comment (nefrob) [inline CONTRIBUTING.md:35]: How should `requirements/compile.py` be invoked?

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,7 @@
 /dist/
 /docs/_build/
 /example/.venv/
+venv
+.python-version
```
Comment (nefrob) [inline .gitignore:11]: Basic `venv` and `pyenv` ignores.

File: tox.ini [hunk 3]
```diff
@@ -17,3 +18,10 @@ commands =
       -W ignore:'pkg_resources is deprecated as an API':DeprecationWarning \
       -m coverage run \
       -m pytest -p no:randomly {posargs:tests}
+
+[testenv:lint]
+description = run linters
+deps =
+    pre-commit
```
Comment (nefrob) [inline tox.ini:25]: We could pin the version here or use the `py311.txt` requirements file. If pinning here then we can drop `pre-commit` as a dependency above.

File: requirements/requirements.in [hunk 4]
```diff
@@ -3,5 +3,7 @@ factory_boy
 faker
 importlib-metadata >= 3.6.0 ; python_version < "3.10"
 numpy
+pip-tools
+pre-commit
```
Comment (nefrob) [inline requirements/requirements.in:7]: Adding these for now since I'm unsure if they should be used for lint and compile commands.

**Comment by adamchainz** [comment]
I do want to add a contributing guide but it would be universal for [all my projects](https://adamj.eu/projects). This PR is changing the processes in ways I don’t want, and I don’t have time to review it properly, so I’m gonna close it.

**Pull request closed without merging**